src/main/java/net/minecraftforge/fml/DatagenModLoader.java

```diff
@@ -39,13 +40,13 @@ public static boolean isRunningDataGen() {
         return runningDataGen;
     }
 
-    public static void begin(final Set<String> mods, final Path path, final java.util.Collection<Path> inputs, Collection<Path> existingPacks, final boolean serverGenerators, final boolean clientGenerators, final boolean devToolGenerators, final boolean reportsGenerator, final boolean structureValidator, final boolean flat) {
+    public static void begin(final Set<String> mods, final Path path, final Collection<Path> inputs, Collection<Path> existingPacks, final String assetIndex, final File assetsDir, final boolean serverGenerators, final boolean clientGenerators, final boolean devToolGenerators, final boolean reportsGenerator, final boolean structureValidator, final boolean flat) {
```

Comment:
Although run from datagen, this is a binary break. Please just leave the old contructor, and deprecate it. Also, please add the new parameters to the end of the parameter list